### ISSUE TITLE
[request] API Calls for specific repo groups

### ISSUE THREAD

**Issue by Nebrethar** [chaoss/augur#397]
**Is your feature request related to a problem? If so, please describe the problem:**
When I click on a repo group on the current `dev` version of augur, I can watch the backend and see that all of `/api/unstable/repos` is called to load a single repo group. This is okay on osshealth.io because the database is quickly queried, but it takes several seconds on my machine. This is where it could be almost instantaneous.

**Describe the solution you'd like:**
I would like to help split up the query into several queries specific to repo-groups.

**Describe alternatives you've considered:**
We could leave it as it is, but it makes the page-loading much less efficient than it could be.

**Additional context: This is the result of looking around at repo-groups**
![image](https://user-images.githubusercontent.com/34951354/65980546-ff113800-e43c-11e9-97a2-4f7eb4e0c7e7.png)

**Comment by percygautam** [comment]
@Nebrethar  I'd like to work on this issue. Can you please explain the problem?

**Comment by Nebrethar** [comment]
Hi @percygautam :)

Thanks for looking at this issue. I will explain to the best of my knowledge.

When I load augur's "repos" page, it makes a call to localhost:5000/api/unstable/repos and that API loads all of Augur's repos at once. That makes sense so it can display all of Augur's stores repositories on one page.

Sometimes, I want to narrow it down to just one repo group's repositories by selecting the "repo groups" page and clicking on a single repo group. A good example is going [here](http://augur.osshealth.io/) and clicking on Zephyrproject-RTOS.

The problem is that by narrowing it down to one repo group, the same API is still called that loads all the repositories in augur. This is true even though only a few repositories will be displayed.

I haven't really dug into it, but it looks like you can use the Group Metric section at lines 379-384 of [server.py](https://github.com/chaoss/augur/blob/dev/augur/server.py). I will add the code here:
```
    def addRepoGroupMetric(self, function, endpoint, **kwargs):
        """Simplifies adding routes that accept repo_group_id"""
        endpoint = f'/{self.api_version}/repo-groups/<repo_group_id>/{endpoint}'
        self.app.route(endpoint)(self.routify(function, 'repo_group'))
        kwargs['endpoint_type'] = 'repo_group'
        self.updateMetricMetadata(function, endpoint, **kwargs)
```
These metrics are built out of [repo_meta.py](https://github.com/chaoss/augur/blob/dev/augur/metrics/repo_meta/repo_meta.py) so a good start would be to figure outn if the API already exists somewhere in there to use it for this purpose.

Let me know what questions you may have! Thanks again for looking at this issue.

**Comment by Nebrethar** [comment]
Also, that code I shared includes the syntax for calling such an API. on my machine (and most) it can be simplified to:
```localhost:5000/repo-groups/<repo_group_id>/{endpoint}'```

@percygautam This is important too:
While I was toying around I found out that 25151 is the zephyr repo_group_id and repos is an existing endpoint under that group of APIs.

So if I call [http://augur.osshealth.io:5000/api/unstable/repo-groups/25151/repos](http://augur.osshealth.io:5000/api/unstable/repo-groups/25151/repos)

This API will return what we need for any repo group there. Now I'm going to task you with finding a way to integrate that change to the frontend.

**Comment by percygautam** [comment]
Hey @Nebrethar , Thanks for assigning me this issue and sorry for the late reply. Being new to the library and all, I am getting some issues with running the server (something related to augur.config.json parsing error). I am working on it. Also, the references you provided of [osshealth](http://augur.osshealth.io/repos/) is not working. Can you please see to this?
I have gained a good overview of the app and am working on the problem.

**Comment by Nebrethar** [comment]
Hi @percygautam! Thanks for the help.

If you don't have any database connection set up on runtime, it will present an error citing augur.config.json parsing (in my experience). If this is your roadblock, I'd suggest raising a separate issue to get the database set up.

If you have a database connection, that error normally means you have an incompatible version of the augur.config.json. I'd suggest raising an issue if the automatically generated one did not work. the file is changed often and the generation script could be outdated.

In other words, the best way to reach the team is with an issue :)

I think the server was not quite working around the time you posted the response. I had checked as soon as you responded and the server refused API connection, but it seems to be working again as of now. I will include a link to the output in the case that it continues not to work for you:

https://pastebin.com/raw/3sjETv5D

I'm glad to continue to help, so let me know any other questions you have related to this. Good luck!

**Comment by mrsaicharan1** [comment]
Hi, As this issue has been inactive for a really long time, I'd like to work on it :D. Would appreciate it if you could assign me to this.

**Comment by mrsaicharan1** [comment]
This seems to be a real problem as it's taking 8-10 seconds just to load one repo group, which is inefficient as such.

**Comment by Nebrethar** [comment]
> Hi, As this issue has been inactive for a really long time, I'd like to work on it :D. Would appreciate it if you could assign me to this.

Done! Thanks for taking this on.

**Comment by mrsaicharan1** [comment]
@Nebrethar I've been digging into the issue but I'm not exactly sure how we could use the aforementioned `addRepoGroupMetric` for this purpose. By clicking on a specific repo group name, it's hitting /api/unstable/repos.


In `addRepoGroupMetric`, It registers the endpoint with the correct one i.e
endpoint = f'/{self.api_version}/repo-groups/<repo_group_id>/{endpoint}'

Could you please help me out here?

**Comment by ccarterlandis** [comment]
If I'm understanding the request correctly, this functionality already exists at the `/api/unstable/repo-groups/<repo_group_id>/repos` endpoint. For example: `http://localhost:5000/api/unstable/repo-groups/10/repos` will return all the repos for only the given `repo_group_id`, which is 10 in this case. 

It would be nice if the frontend took advantage of this, but I think that's outside the scope of this issue since this is about the endpoint request. As such I'm going to close the issue, but if anyone disagrees please feel free to reopen 😄
